### ISSUE TITLE
lsp4j update: Allow server implementations to discover what client make a given request

### ISSUE THREAD

**Issue by robstryker** [redhat-developer/rsp-server#187]
https://github.com/eclipse/lsp4j/pull/260

A PR was approved and merged into lsp4j to allow a more unified API to determine which client is making a given request. While rsp-server has it's own implementation of this feature, once the PR is merged (it is) and released (not yet) we should consume this version of lsp4j and update our code to make use of it. 

Also, when we update to that version of lsp4j, we might also want to fix https://github.com/redhat-developer/rsp-server/issues/152  which should allow us to stop "bundling" lsp4j.jsonrpc and gson, and instead installing those bundles into the osgi runtime directly.

**Comment by robstryker** [comment]
0.6.0 Does not include this fix. Not much action in lsp4j since then. This issue must be delayed.

**Comment by robstryker** [comment]
We use lsp4j 0.7.0 now

commit e61d89710e3f82fd3937322e0239595d36d97296
Author: Rob Stryker <rob@oxbeef.net>
Date:   Thu Mar 7 12:54:30 2019 -0500
